fix(scss): scss config vars, missing imports and hotfixes

hellobrian opened the pull request:
## Overview

resubmission of #72 
related https://github.com/carbon-design-system/carbon-components/issues/71
resolves https://github.ibm.com/Bluemix/bluemix-components-react/issues/610

### Add `_css--typography.scss`

This is all of our `h1`, `h2`, `h3` etc styles that used to be part of `_css--body.scss`.
This new SCSS file is controlled by a **new variable**: `$css--typography: true !default;`

### New defaults for global SCSS variables

```scss
$css--font-face: false !default;
$css--helpers: true !default;
$css--body: true !default;
$css--reset: true !default;
$css--typography: true !default;
```

All variables default to `true` except for `$css--font-face: false !default;`.

### New global SCSS variable conditionals

```scss
@if global-variable-exists('css--body') == false or $css--body == true {
  @include css-body;
}
```
Conditionals have changed.

CSS output occurs when:
- if the `$var` **does not** exist, then output CSS.
- if the `$var` equals `true`, then output CSS

The user can opt out of CSS output by setting the variable to `false`.
- `$css--body: false;` will prevent CSS output. 

### 🔥 Fixes Fixes Fixes (so many fixes, 🙏 🇨🇦)

- Add missing `@import` statements (tested for importing single components w/o vars).
- Remove `<use>` and inline SVG icons in HTML files
- Change SCSS for various visual fixes, see comments for details.

Commit:
chore(.gitignore): be more specific about ignored files

Commit:
refactor(global-scss-vars): new vars for scss config

hellobrian commented on src/globals/scss/_css--body.scss:
```diff
@@ -4,72 +4,8 @@
-
 @mixin css-body {
-  .bx--body {
+  body {
```
Setting `$css--body: true` will output these `body` styles to CSS. 
This is the default.

hellobrian commented on src/globals/scss/_css--helpers.scss:
```diff
@@ -13,6 +13,14 @@
     white-space: nowrap;
   }
+
+  .bx--body {
```
`.bx--body` class gets applied when `$css--helpers: true`, which is the same styles as `body` from `_css--body.scss`

hellobrian commented on src/globals/scss/styles.scss:
```diff
@@ -6,6 +6,7 @@ $css--font-face: false !default;
 $css--helpers: true !default;
 $css--body: true !default;
 $css--reset: false !default;
+$css--typography: true !default;
```
`$css--typography: true` will output CSS from `_css--typography.scss` and this is default.

Commit:
fix(scss): all _css--* files @import reset now

tw15egan commented on src/globals/scss/_css--typography.scss:
```diff
@@ -0,0 +1,76 @@
+@import 'vars';
+@import 'mixins';
+@import 'typography';
+@import 'reset';
```
I think this should be `css--reset`

hellobrian replied:
blargh! sorry, thanks for pointing this out haha

tw15egan commented on src/globals/scss/_css--helpers.scss:
```diff
@@ -1,3 +1,5 @@
+@import 'reset';
```
I think this should be `css--reset`

tw15egan commented on src/globals/scss/_css--body.scss:
```diff
@@ -2,74 +2,11 @@
 @import 'vars';
 @import 'mixins';
 @import 'typography';
+@import 'reset';
```
I think this should be `css--reset`

tw15egan reviewed (changes requested):
👍 ✅ ❗️

Commit:
fix(SCSS): fix import type for css--reset

Commit:
fix(styles.scss): set reset to true by default

Commit:
fix(scss-global-vars): change conditionals

- if var does not exist, then output CSS
- OR if var equals true, then output CSS
- User must set var equal to false to prevent output CSS

Commit:
fix(reset.scss): change conditional

- always ouputs CSS from one of the reset mixins when var does not exist
- if var equals false, then add reset CSS to each class selector
- if var equals true, then output reset CSS once at top and no where
  else

Commit:
fix(copy-button): add inline svg for icon

Commit:
chore(button): move import reset to last

Commit:
fix(file-uploader): fixes for importing by itself

- imports loading component
- fix styles for filename nodes

Commit:
fix(checkbox): add misssing form import

Commit:
fix(radio): add missing form import

Commit:
fix(toggle): add missing form import

Commit:
fix(search): center icons in buttons

Commit:
fix(inline-select): use inline svg

Commit:
fix(link): add missing import helper-mixins

Commit:
fix(number-input): add missing imports

Commit:
fix(code-snippet): add missing import

Commit:
fix(card): make scrollbars invisible

Commit:
style(mixins): remove space between first and 2nd line

Commit:
fix(modal): remove overflow-y: auto

Commit:
fix(inline-notification): add spacing between close button and text

Commit:
fix(tooltip): missing icons and fonts

Commit:
fix(iln): remove unneeded reset mixins

Commit:
fix(accordion): normalize icon size w/padding

Commit:
fix(progress-indicator): fix line spacing

Commit:
fix(toolbar): inline svg and new focus state

hellobrian commented on src/components/accordion/_accordion.scss:
```diff
@@ -38,8 +39,8 @@
-    height: .75rem;
-    width: .75rem;
+    height: 1.25rem;
+    width: 1.25rem;
```
Change: `padding` was affecting size of accordion arrow icons making them appear smaller than normal.

hellobrian commented on src/components/card/_card.scss:
```diff
@@ -85,17 +81,21 @@
     @include text-overflow(rem(180px));
     font-weight: 400;
     margin: 0;
+    line-height: 1.2;
```
Change: new styles here fix unwanted scrollbar from appearing

![image](https://cloud.githubusercontent.com/assets/4185382/26649154/8e03f9ae-460a-11e7-88fb-1f0d22355557.png)

☝️ with unwanted scrollbar

hellobrian commented on src/components/file-uploader/_file-uploader.scss:
```diff
@@ -58,20 +55,19 @@
+    align-items: center;
     color: $text-01;
     margin-right: 1rem;
+    height: 1.875rem;
```
Change: styles added here fix unwanted scrollbar and prevents it from appearing

hellobrian commented on src/components/notification/_inline-notification.scss:
```diff
@@ -17,6 +17,7 @@
     @include font-smoothing;
     @include font-size('14');
     display: flex;
+    justify-content: space-between;
```
Change: fixes missing space between inline-notification text and close button.

![image](https://cloud.githubusercontent.com/assets/4185382/26649256/f3b1a08a-460a-11e7-9b16-faa1522b2c5a.png)

☝️  missing space

hellobrian commented on src/components/progress-indicator/_progress-indicator.scss:
```diff
@@ -25,7 +25,7 @@
     position: absolute;
     top: .625rem;
-    right: 99.5%;
+    right: 100%;
```
Change: fixes gap between line and indicators.

hellobrian commented on src/components/search/_search.scss:
```diff
@@ -101,6 +101,7 @@
     fill: $brand-01;
     height: 1rem;
     width: 1rem;
+    vertical-align: middle;
```
Need to add more styles

```css

    position: relative;
    top: -2px;
```

hellobrian commented on src/components/toolbar/_toolbar.scss:
```diff
@@ -93,6 +93,10 @@ $css--helpers: true;
     width: rem(32px);
+
+    &:focus {
+      @include focus-outline;
```
Change: fixes missing focus outline on toolbar search button

hellobrian commented on src/components/tooltip/_tooltip.scss:
```diff
@@ -12,6 +13,7 @@
   }
 
   .bx--tooltip__trigger {
+    @include helvetica;
```
Change: adds Helvetica font-family to tooltip triggers

Commit:
fix(search): center icons in buttons

hellobrian commented on src/components/search/_search.scss:
```diff
@@ -94,9 +94,14 @@
+    position: relative;
+    top: -1px;
+    box-sizing: border-box;
+    vertical-align: middle;
```
Change: centers icon in buttons

Commit:
fix(file-uploader): import css--typography

Commit:
fix(card): import css--typography

Commit:
fix(data-table): import css--typography

Commit:
fix(detail-page-header): import css--typography

Commit:
fix(dropdown): import css--typography

Commit:
fix(modal): import css--typography

Commit:
fix(module): import css--typography

Commit:
fix(notifications): import css--typography

Commit:
fix(accordion): import css--typography

Commit:
fix(breadcrumb): import css--typography

Commit:
fix(footer): import css--typography

Commit:
fix(components): import css--typography

- iln
- tabs
- tooltip
- unified-header

hellobrian commented:
@chrisdhanaraj Went through the components one more time and made sure all components that need `_css--typography.scss` are now importing it.

chrisdhanaraj commented:
👍 ✅
